assets/app/data/services/upgrade/upgrade-admin.factory.js

```diff
@@ -0,0 +1,33 @@
+(function() {
+
+    angular
+        .module('crowbarData.upgrade')
+        .factory('upgradeUpgradeAdminFactory', upgradeUpgradeAdminFactory);
+
+    upgradeUpgradeAdminFactory.$inject = ['$q', '$http'];
+    /* @ngInject */
+    function upgradeUpgradeAdminFactory($q, $http) {
+        return {
+            getAdminUpgrade: getAdminUpgrade,
+            getAdminUpgradeStatus: getAdminUpgradeStatus
+        };
+
+        function getAdminUpgrade() {
+            var requestOptions = {
+                method: 'POST',
+                url: '/api/upgrade7/admin-upgrade'
+            };
+
+            return $http(requestOptions);
+        }
+
+        function getAdminUpgradeStatus() {
```

Comment:
I think we should rely on the Crowbar entity to determine the status of the admin node upgrade.
Actually, that's what we need to use to determine the status of the whole Upgrade flow whenever the browser is opened.